<launch>
    <!-- YOLO with DriveWorks Camera Launch File -->
    
    <!-- Arguments for easy configuration -->
    <arg name="model_path" default="$(find drive_yolo)/models/yolov8n.engine" />
    <arg name="confidence_threshold" default="0.1" />
    <arg name="detection_threshold" default="0.1" />
    <arg name="quiet_mode" default="true" />
    <arg name="camera_params" default="camera-name=F008A120RM0AV2,interface=csi-ab,link=1,output-format=processed" />
    
    <!-- YOLO Detection Node with DriveWorks -->
    <node name="drive_yolo_node" pkg="drive_yolo" type="drive_yolo_node" output="screen">
        <!-- Model configuration -->
        <param name="model_path" value="$(arg model_path)" />
        
        <!-- Camera configuration -->
        <param name="use_driveworks_camera" value="true" />
        <param name="camera_params" value="$(arg camera_params)" />
	    <param name="enable_hood_filtering" value="true"/>
        <param name="ignore_hood_pixels" value="290"/>
        <param name="enable_hood_filtering" value="true"/>
        <param name="vehicle_confidence_threshold" value="0.5" />      <!-- 50% for vehicles -->
        <param name="pedestrian_confidence_threshold" value="0.2" />   <!-- 20% for pedestrians -->
            
        <!-- Detection thresholds -->
        <param name="confidence_threshold" value="$(arg confidence_threshold)" />
        <param name="detection_threshold" value="$(arg detection_threshold)" />
        <param name="quiet_mode" value="$(arg quiet_mode)" />
    </node>
    
</launch>
